spells.cpp

```diff
@@ -489,31 +448,33 @@ void Game::ProcessCastGateLore(Unit *u, parser::string_parser& parser, orders_ch
         if (checker) return;
 
         TeleportOrder *order;
-        if (u->teleportorders && u->teleportorders->spell == S_GATE_LORE && u->teleportorders->gate == -1 ) {
-            order = u->teleportorders;
-        } else {
-            order = new TeleportOrder;
-            order->gate = -1;
-            order->spell = S_GATE_LORE;
-            order->level = 1;
-            u->ClearCastOrders();
-            u->teleportorders = order;
-        }
+        int level = 1;
+        int gate = -1;
 
         token = parser.get_token();
         if (token == "level") {
-            order->gate = -2;
-            order->level = 2;
+            gate = -2;
+            level = 2;
             token = parser.get_token();
         }
         if (token != "units") return;
```

Comment:
This line needs to move below the block below, otherwise it exits early.  This is the most likely cause of breaking the snapshot tests.
```suggestion
```

Reply:
I've reverted my change, I didn't realize this was intended behavior. Also made the snapshot tests run with the CMake executables and cleaned up the shell scripts a little bit in the process.

Reply:
> If it does, then the TeleportOrder is still created in line 467, even though it has no units to act upon. Is that really desired? Alternatively, if another teleport order already exist, its level and gate values get modified.

Yeah.  If you issue two contradictory forms of the teleport order, that last one given is the one that is effective.  And teleport always acts on self, only certain forms of it use units as well, and those versions are higher level. Thelogic in here is correct albeit confusing, especially on first readthrough